hudi-common/src/main/java/org/apache/hudi/metadata/FileSystemBackedTableMetadata.java

```diff
@@ -49,12 +60,48 @@ public FileSystemBackedTableMetadata(SerializableConfiguration conf, String data
 
   @Override
   public List<String> getAllPartitionPaths() throws IOException {
-    FileSystem fs = new Path(datasetBasePath).getFileSystem(hadoopConf.get());
     if (assumeDatePartitioning) {
+      FileSystem fs = new Path(datasetBasePath).getFileSystem(hadoopConf.get());
       return FSUtils.getAllPartitionFoldersThreeLevelsDown(fs, datasetBasePath);
-    } else {
-      return FSUtils.getAllFoldersWithPartitionMetaFile(fs, datasetBasePath);
     }
+
+    List<Path> pathsToList = new LinkedList<>();
+    pathsToList.add(new Path(datasetBasePath));
+    List<String> partitionPaths = new ArrayList<>();
+
+    // TODO: Get the parallelism from HoodieWriteConfig
+    final int fileListingParallelism = 1500;
```

Comment:
+1 there can we pass this in as an arg to `FileSystemBackedTableMetadata` class? This default is too much for query side.

Reply:
This is the max. There is a Math.min below.

Reply:
Yeah 1500 is just the maximum. The only downside with this current code will be that one cannot increase parallelism beyond 1500. These classes already have too many parameters and adding a new one here again results in changing all the consumers (and their consumers) which again affects more and more files.

I think it would be better that in a separate PR, we should get rid of these individual parameters and have the consumers pass in HoodieMetadataConfig (which has all these params).

Reply:
Okay my bad. can we pull this into a static final variable for now.

Reply:
Will do.